packages/azure-themes/src/azure/styles/MessageBar.styles.ts

```diff
@@ -0,0 +1,266 @@
+import { IExtendedSemanticColors } from '../IExtendedSemanticColors';
+
+const IconButtonStyles = (props: IMessageBarStyleProps): IStyle => {
+  const { theme, messageBarType } = props;
+  const semanticColors = theme.semanticColors as IExtendedSemanticColors;
+
+  return [
+    (messageBarType === MessageBarType.error || messageBarType === MessageBarType.severeWarning) && {
+      backgroundColor: semanticColors.statusErrorBackground,
+      selectors: {
+        '.ms-Button-icon': {
+          color: semanticColors.statusErrorText,
+          selectors: {
+            '&:hover': {
+              backgroundColor: semanticColors.statusErrorBackground
+            },
+            '&:active': {
+              backgroundColor: semanticColors.statusErrorBackground
+            }
+          }
+        }
+      }
+    },
+
+    messageBarType === MessageBarType.info && {
+      backgroundColor: semanticColors.statusInformationBackground,
+      selectors: {
+        '.ms-Button-icon': {
+          color: semanticColors.statusInformationText,
+          selectors: {
+            '&:hover': {
+              backgroundColor: semanticColors.statusInformationBackground
+            },
+            '&:active': {
+              backgroundColor: semanticColors.statusInformationBackground
+            }
+          }
+        }
+      }
+    },
+
+    messageBarType === MessageBarType.success && {
+      backgroundColor: semanticColors.statusSuccessBackground,
+      selectors: {
+        '.ms-Button-icon': {
+          color: semanticColors.statusSuccessText,
+          selectors: {
+            '&:hover': {
+              backgroundColor: semanticColors.statusSuccessBackground
+            },
+            '&:active': {
+              backgroundColor: semanticColors.statusSuccessBackground
+            }
+          }
+        }
+      }
+    },
+
+    (messageBarType === MessageBarType.warning || messageBarType === MessageBarType.blocked) && {
+      backgroundColor: semanticColors.statusWarningBackground,
+      selectors: {
+        '.ms-Button-icon': {
+          color: semanticColors.statusWarningText,
+          selectors: {
+            '&:hover': {
+              backgroundColor: semanticColors.statusWarningBackground
+            },
+            '&:active': {
+              backgroundColor: semanticColors.statusWarningBackground
+            }
+          }
+        }
+      }
+    },
+
+    !messageBarType && {
+      backgroundColor: semanticColors.bodyBackground,
+      selectors: {
+        '.ms-Button-icon': {
+          color: semanticColors.bodyText,
+          selectors: {
+            '&:hover': {
+              backgroundColor: semanticColors.buttonBackgroundHovered
+            },
+            '&:active': {
+              backgroundColor: semanticColors.buttonBackgroundHovered
+            }
+          }
+        }
+      }
+    }
+  ];
+};
+
+const RootStyles = (props: IMessageBarStyleProps): IStyle => {
```

Comment:
nit: Seems like `RootStyles` along with `IconButtonStyles` have a lot in common and differ only the former having `color` added. Consider maybe extracting these variants into one place and not having redundant repetition.

Reply:
yes -- we'll look into streamlining this one.